### DISCUSSION TITLE
cmus fails to open audio device .

### DISCUSSION THREAD

**Discussion by neelchakrabrty** [cmus/cmus#1087] [General]
---

1) `uname -a`:
`Linux  5.5.3-gentoo 
`
2) `cmus --version`:
`cmus v2.8.0`

3) `cmus --plugins`:
```
Input Plugins: /usr/lib64/cmus/ip
  mad:
    Priority: 55
    File Types: mp3 mp2
    MIME Types: audio/mpeg audio/x-mp3 audio/x-mpeg
  aac:
    Priority: 50
    File Types: aac
    MIME Types: audio/aac audio/aacp
  flac:
    Priority: 50
    File Types: flac fla
    MIME Types:
  mp4:
    Priority: 50
    File Types: mp4 m4a m4b
    MIME Types:
  wav:
    Priority: 50
    File Types: wav
    MIME Types:
  vorbis:
    Priority: 50
    File Types: ogg oga ogx
    MIME Types: application/ogg audio/x-ogg
  cue:
    Priority: 50
    File Types:
    MIME Types: application/x-cue

Output Plugins: /usr/lib64/cmus/op
  alsa
```
My rc file is 
```
set output_plugin=alsa
set dsp.alsa.device=default
set mixer.alsa.device=default
set mixer.alsa.channel=Master

```
The error says 
`Error: opening audio device: Invalid argument`

How do I fix this ?

**Comment by pgaskin** [comment]
What is the output of `aplay -t raw --dump-hw-params < /dev/null` and `aplay -l`?

**Comment by neelchakrabrty** [comment]
output of `aplay -t raw --dump-hw-params < /dev/null`
```
Playing raw data 'stdin' : Unsigned 8 bit, Rate 8000 Hz, Mono
HW Params of device "default":
--------------------
ACCESS:  MMAP_INTERLEAVED RW_INTERLEAVED
FORMAT:  S16_LE S32_LE
SUBFORMAT:  STD
SAMPLE_BITS: [16 32]
FRAME_BITS: [32 192]
CHANNELS: [2 6]
RATE: [44100 192000]
PERIOD_TIME: (31 185760)
PERIOD_SIZE: [6 8192]
PERIOD_BYTES: [128 196608]
PERIODS: [2 32]
BUFFER_TIME: (62 371520)
BUFFER_SIZE: [12 16384]
BUFFER_BYTES: [128 65536]
TICK_TIME: ALL
--------------------
aplay: set_params:1343: Sample format non available
Available formats:
- S16_LE
- S32_LE

```
output of `aplay -l`
```
**** List of PLAYBACK Hardware Devices ****
card 0: PCH [HDA Intel PCH], device 0: ALC1150 Analog [ALC1150 Analog]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 0: PCH [HDA Intel PCH], device 1: ALC1150 Digital [ALC1150 Digital]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 1: NVidia [HDA NVidia], device 3: HDMI 0 [HDMI 0]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 1: NVidia [HDA NVidia], device 7: HDMI 1 [HDMI 1]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 1: NVidia [HDA NVidia], device 8: HDMI 2 [HDMI 2]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 1: NVidia [HDA NVidia], device 9: HDMI 3 [HDMI 3]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 1: NVidia [HDA NVidia], device 10: HDMI 4 [HDMI 4]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 1: NVidia [HDA NVidia], device 11: HDMI 5 [HDMI 5]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 1: NVidia [HDA NVidia], device 12: HDMI 6 [HDMI 6]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
```

**Comment by mrkapqa** [comment]
helloo ,

i have the same problem whenever i change inside

 ~/.config/cmus/rc

from pulse to alsa

`set output_plugin=alsa`

this lead cmus to 

> Error: opening audio device: No such file or directory<

**Comment by mrkapqa** [comment]
`aplay -l
**** List of PLAYBACK Hardware Devices ****
card 0: Intel [HDA Intel], device 0: CX20561 Analog [CX20561 Analog]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 0: Intel [HDA Intel], device 1: CX20561 Digital [CX20561 Digital]
  Subdevices: 1/1
  Subdevice #0: subdevice #583 `

`aplay -t raw --dump-hw-params < /dev/null
Playing raw data 'stdin' : Unsigned 8 bit, Rate 8000 Hz, Mono
HW Params of device "default":
--------------------
ACCESS:  MMAP_INTERLEAVED MMAP_NONINTERLEAVED MMAP_COMPLEX RW_INTERLEAVED RW_NONINTERLEAVED
FORMAT:  S8 U8 S16_LE S16_BE U16_LE U16_BE S24_LE S24_BE U24_LE U24_BE S32_LE S32_BE U32_LE U32_BE FLOAT_LE FLOAT_BE FLOAT64_LE FLOAT64_BE MU_LAW A_LAW IMA_ADPCM S20_LE S20_BE U20_LE U20_BE S24_3LE S24_3BE U24_3LE U24_3BE S20_3LE S20_3BE U20_3LE U20_3BE S18_3LE S18_3BE U18_3LE U18_3BE
SUBFORMAT:  STD
SAMPLE_BITS: [4 64]
FRAME_BITS: [4 640000]
CHANNELS: [1 10000]
RATE: [4000 4294967295)
PERIOD_TIME: (21333 21334)
PERIOD_SIZE: (85 91628833)
PERIOD_BYTES: (42 4294967295)
PERIODS: (0 17247242)
BUFFER_TIME: [1 4294967295]
BUFFER_SIZE: [170 1466015503]
BUFFER_BYTES: [85 4294967295]
TICK_TIME: ALL`

cmus is version 2.8 and has been compiled with
./configure
make
sudo make install

**Comment by nefthy** [comment]
check ~/.cmus/autosave for the values:

dsp.alsa.device=default
mixer.alsa.device=default
mixer.alsa.channel=PCM
output_plugin=alsa

if they are missing add them
